packages/jest-cli/src/run_jest.js

```diff
@@ -58,7 +58,13 @@ const getTestPaths = async (
   }
 
   const shouldTestArray = await Promise.all(
-    data.tests.map(test => jestHooks.shouldRunTestSuite(test.path)),
+    data.tests.map(test =>
+      jestHooks.shouldRunTestSuite({
+        config: test.context.config,
+        duration: test.duration,
+        testPath: test.path,
```

Comment:
(`testPath` vs `path`) and (`config` vs `projectConfig`):

 I like `testPath` and `config` better, but I'm fine with anything